pkg/printers/checkstyle.go

```diff
@@ -0,0 +1,78 @@
+package printers
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+type checkstyleOutput struct {
+	XMLName xml.Name          `xml:"checkstyle"`
+	Version string            `xml:"version,attr"`
+	Files   []*checkstyleFile `xml:"file"`
+}
+
+type checkstyleFile struct {
+	Name   string             `xml:"name,attr"`
+	Errors []*checkstyleError `xml:"error"`
+}
+
+type checkstyleError struct {
+	Column   int    `xml:"column,attr"`
+	Line     int    `xml:"line,attr"`
+	Message  string `xml:"message,attr"`
+	Severity string `xml:"severity,attr"`
+	Source   string `xml:"source,attr"`
+}
+
+const defaultSeverity = "error"
+
+type Checkstyle struct{}
+
+func NewCheckstyle() *Checkstyle {
+	return &Checkstyle{}
+}
+
+func (Checkstyle) Print(ctx context.Context, issues <-chan result.Issue) (bool, error) {
+	out := checkstyleOutput{
+		Version: "5.0",
+	}
+
+	files := map[string]*checkstyleFile{}
+
+	for issue := range issues {
+		file, ok := files[issue.FilePath()]
+		if !ok {
+			file = &checkstyleFile{
+				Name: issue.FilePath(),
+			}
+
+			files[issue.FilePath()] = file
+		}
+
+		newError := &checkstyleError{
+			Column:   issue.Column(),
+			Line:     issue.Line(),
+			Message:  issue.Text,
+			Source:   issue.FromLinter,
+			Severity: defaultSeverity,
+		}
+
+		file.Errors = append(file.Errors, newError)
+	}
+
+	out.Files = make([]*checkstyleFile, len(files))
```

Comment:
maybe `make([]*checkstyleFile, 0, len(files))`?

Reply:
This kind of syntax is not supported for maps.
It'll give an error `too many arguments to make(map)`
https://play.golang.org/p/b7b5e8jgEVp

Reply:
But it’s a slice, not map

Reply:
oh, crap. yes, you're right :)